### PR TITLE
CIAPP-7065: CI Visiility: use include files for repeated content

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
### What does this PR do?
The following sections are repeated across multiple documentation files for CI test visibility:
- Github metadata upload instructions
- Use of Datadog agent for on-premises users
- Information collected (except for ITR-related notes, but eventually that will also be repeated)

There are a couple of bits worth mentioning:
- the Swift blurb about Git metadata is slightly different (the bigger difference is missing a sentence linking to CI-provider envs)
- the JUnit upload Git metadata blurb also changes slightly, dropping mention of the Datadog CLI

Note: this is the first use of the include-markdown shortcode in some time

### Motivation
This cuts down on repetitive changes across multiple pages that include the same content.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by romainkomorndatadog** [comment]
I think this is, overall, a good change, but it does introduce a bit of rigidity in how we publish information, or might result in some inconsistencies across tracers (eg: if we change the Git metadata uploaded by one of the languages' tracers).

**Comment by romainkomorndatadog** [comment]
@alai97 , I feel like I should've run this idea by you before I started. I'm totally happy to abandon it if you feel like it's a step backwards in terms of how easy it is to maintain these docs.

**Review by juan-fernandez — APPROVED** [review]
this looks like a nice change to me. Only thing that I'd make sure of is that the links are still correct (we're removing some of the links when using `<include-markdown>`) and that there aren't any non used links at the bottom of each language's markdown file.

In the case of javascript and junit, the links seem correct, but I haven't checked every language

**Comment by kayayarai** [comment]
@romainkomorndatadog Hi! This is pretty cool, but we already have an established way of doing content reuse in the docs, which is documented here: https://datadoghq.atlassian.net/wiki/spaces/docs4docs/pages/1182958054/Markdown#General-content-reuse-shortcodes . I'll let @alai97 weigh in, but I think you'll want to restructure this PR according to those guidelines. This ensures links work and that it gets translated and maintained properly. Gonna put a WIP label on this PR.

**Comment by romainkomorndatadog** [comment]
@kayayarai Oh, perfect. I'll take a look at that. Somehow that wiki page wasn't on my list of docs to check, but it is now.

@alai97 , if you feel like the idea of this PR's good to keep going, I'll update the shortcodes I'm using.

**Comment by alai97** [comment]
Yes please @romainkomorndatadog, just for easy reference, the reusable shortcode in https://github.com/DataDog/documentation/blob/master/content/en/synthetics/api_tests/http_tests.md#select-locations for example lives in https://github.com/DataDog/documentation/blob/master/layouts/shortcodes/managed-locations.md!

File: content/en/continuous_integration/tests/junit_upload.md [hunk 1]
```diff
@@ -230,53 +230,7 @@ Additionally, configure the Datadog site to use the selected one ({{< region-par
 
 ## Collecting repository and commit metadata
 
-Datadog uses Git information for visualizing your test results and grouping them by repository and commit. Git metadata is collected by the Datadog CI CLI from CI provider environment variables and the local `.git` folder in the project path, if available. To read this directory, the [`git`][8] binary is required.
```
[the remark this answers is not in the document]
Comment (alai97) [reply content/en/continuous_integration/tests/junit_upload.md:233]: I removed it, thanks for flagging!

**Comment by romainkomorndatadog** [comment]
Refactored to use the `{{% reuse/... %}}` shortcode way, although we appear to be the only people using that method in the repo.

There are a couple of quirks:
- I had to remove the region-based shortcode in the `DD_SITE` env var under agentless configuration (I don't think this is a significant loss)
- when mixing the shortcode with other text, empty lines are necessary to force wrapping the text in a `<p>` tag (otherwise it just leaves things in the wrapping `<div>` and ends up making spacing look odd)

**Comment by romainkomorndatadog** [comment]
Closing in favor of https://github.com/DataDog/documentation/pull/18843/files

**Pull request closed without merging**